arches/app/search/components/search_results.py

```diff
@@ -72,6 +73,15 @@ def append_dsl(
             GeoBoundsAgg(field="points.point", name="bounds")
         )
         nested_agg.add_aggregation(nested_agg_filter)
+
+        if self.user and self.user.id:
+            search_query = Bool()
+            subsearch_query = Bool()
+            # TODO: call to permissions framework with subsearch_query
```

Comment:
This is a placeholder, but presumably the next step would be to add a method to the permissions frameworks to do something/nothing with the user.

(As a side-note, on our fork, we have tried to get a little more separation between the request and the business logic - ideally we would not hand the full request object to the permissions framework, but the user and perhaps a parameter dictionary; does that sound reasonable?)

Reply:
It may also be prudent to do the principal user "subquery" logic in the permissions framework too, so it just gets the search_query back, as there are definitely cases where the principal_user should not have access to some or all of a resource. (e.g. an application form)